### PR TITLE
Changed C++20 to emit C++20 instead of C++Latest in MSVC

### LINKED ISSUE #1746: cppdialect c++20 outputs c++latest on VS2019
I use cppdialect "C++20" currently, which resolves to `/std:c++latest` but in my version of vs2019 I want it to resolve to `/std:c++20`.

Imported from this comment: https://github.com/premake/premake-core/issues/971#issuecomment-962028454

### PR DESCRIPTION
**Pull request by nickclark2016**
**What does this PR do?**

Changes C++20 to emit `stdcpp20` instead of `stdcpplatest` in VS projects VS2017 or newer.  Adds `c11` and `c17` outputs to VS 2019 and newer projects.

**How does this PR change Premake's behavior?**

Existing projects with the `cppdialect` set to `C++20` will now emit `stdcpp20` instead of `stdcpplatest` for VS 2017 and newer.

**Anything else we should know?**

Closes #1746

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [x] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: modules/vstudio/vs2010_vcxproj.lua [hunk 1]
```diff
@@ -1477,9 +1477,15 @@
 			elseif (cfg.cppdialect == "C++17") then
 				m.element("LanguageStandard", nil, 'stdcpp17')
 			elseif (cfg.cppdialect == "C++20") then
-				m.element("LanguageStandard", nil, 'stdcpplatest')
+				m.element("LanguageStandard", nil, 'stdcpp20')
```
Comment (samsinsane) [inline modules/vstudio/vs2010_vcxproj.lua:1480]: According to the docs, this is only available in VS2019?

File: modules/vstudio/vs2010_vcxproj.lua [hunk 2]
```diff
@@ -1477,9 +1477,15 @@
...
 			elseif (cfg.cppdialect == "C++latest") then
 				m.element("LanguageStandard", nil, 'stdcpplatest')
+			elseif _ACTION >= "vs2019" then
+				if (cfg.cdialect == "C11") then
+					m.element("LanguageStandard_C", nil, 'stdc11')
+				elseif (cfg.cdialect == "C17") then
+					m.element("LanguageStandard_C", nil, 'stdc17')
+				end
```
Comment (samsinsane) [inline modules/vstudio/vs2010_vcxproj.lua:1488]: These should be in their own function independent of `cppdialect` - this change currently means specifying both `cppdialect` and `cdialect` will result in only the C++ standard to be emit by Premake.

File: modules/vstudio/vs2010_vcxproj.lua [hunk 3]
```diff
@@ -1539,7 +1545,7 @@
 			elseif (cfg.cppdialect == "C++17") then
 				table.insert(opts, "/std:c++17")
 			elseif (cfg.cppdialect == "C++20") then
-				table.insert(opts, "/std:c++latest")
+				table.insert(opts, "/std:c++20")
```
Comment (samsinsane) [inline modules/vstudio/vs2010_vcxproj.lua:1548]: Similar to the comment above, VS2015 likely doesn't support this flag.

File: src/_premake_init.lua [hunk 4]
```diff
@@ -757,6 +757,7 @@
 			"C90",
 			"C99",
 			"C11",
+			"C17",
```
Comment (samsinsane) [inline src/_premake_init.lua:760]: This should probably be done in a separate PR - I think it's far more likely `gmake2` users are going to want to use this than VS users. Similarly, there should be the `gnu17` option too.

**Review by samsinsane — CHANGES_REQUESTED** [review]

File: src/_premake_init.lua (hunk 4 above)
Comment (nickclark2016) [reply src/_premake_init.lua:760]: I was just getting all the language flags for MSVC up to date in their mappings. I'd be happy to do gmake2 in a separate PR.

Comment (samsinsane) [reply src/_premake_init.lua:760]: Yeah that's fair enough, I'll resolve this comment.

**Review by samsinsane — APPROVED** [review]
